### ISSUE TITLE
NumpyGenericOperator lincomb apply_inverse infinite recursion in 0.5.x

### ISSUE THREAD

**Issue by pmli** [pymor/pymor#794]
The following raises `RecursionError` on 0.5.x.
```python
import numpy as np
from pymor.operators.numpy import NumpyMatrixOperator, NumpyGenericOperator

n = 2
I1 = NumpyMatrixOperator(np.eye(n))
I2 = NumpyGenericOperator(lambda x: x, dim_source=n, dim_range=n)
v = I1.source.from_numpy(np.ones(n))

(I1 + I2).apply_inverse(v)
```

On master, `jacobian` method raises `NotImplementedError`.

Should the recursion error be fixed in 0.5.x?

**Comment by sdrave** [comment]
If it can be trivially fixed, yes. Otherwise, I would say that the next release is too near for investing work into fixing this error.

**Comment by pmli** [comment]
Fixed in #810.
